File: java/sprykerFileCreator/model/definition/spryker/ZedConfigDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedConfigDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_CONFIG;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}Config";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}";
    }

    @Override
    public String getMethodForReturnType() {
        return "getConfig()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {};

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Kernel\\AbstractBundleConfig";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/definitions/spryker/ZedConfigDefinition.kt
package com.github.tobi812.sprykerplugin.services.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.services.definitions.ClassDefinitionInterface

class ZedConfigDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_CONFIG
    override val namePattern: String
        get() = "{{ bundleName }}Config"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}"
    override val methodForReturnType: String
        get() = "getConfig()"
    override val docBlockClasses: Array<String>
        get() = arrayOf()
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Kernel\\AbstractBundleConfig"
}
